### PR TITLE
[NVPTX] Use PTX prmt for llvm.bswap

### PR DESCRIPTION
**Pull request by AlexMaclean**

### CODE REVIEW

File: llvm/test/CodeGen/NVPTX/bswap.ll [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+; CHECK-LABEL: @bswap64
+; CHECK:    ld.param.u64    %rd1, [bswap64_param_0];
+; CHECK:    { .reg .b32 tmp; mov.b64 {%r1, tmp}, %rd1; }
+; CHECK:    prmt.b32        %r2, %r1, 0, 291;
+; CHECK:    { .reg .b32 tmp; mov.b64 {tmp, %r3}, %rd1; }
+; CHECK:    prmt.b32        %r4, %r3, 0, 291;
+; CHECK:    mov.b64         %rd2, {%r4, %r2};
+; CHECK:    st.param.b64    [func_retval0+0], %rd2;
+
```
Comment (Artem-B) [inline llvm/test/CodeGen/NVPTX/bswap.ll:45]: Is that expected that we're ending up with a single `PRMT` instruction in SASS?
godbolt.org/z/KoMc679rE
```
 PRMT R4, R5, 0x123, RZ
 RET
```

There's no good documentation on how exactly PRMT works on SASS level, but I would expect to see two of them.

Comment (AlexMaclean) [reply llvm/test/CodeGen/NVPTX/bswap.ll:45]: In the above link it looks like the return type is `.b8 func_retval0[2]` which is small enough that some optimization is occurring in `ptxas`. When the return type is 8 bytes long 2 `PRMT` instructions are generated [https://godbolt.org/z/WPjrfcM98](https://godbolt.org/z/WPjrfcM98)

Comment (Artem-B) [reply llvm/test/CodeGen/NVPTX/bswap.ll:45]: Oh. I've missed the return type part.

File: llvm/test/CodeGen/NVPTX/bswap.ll [hunk 2]
```diff
@@ -0,0 +1,54 @@
+; RUN: llc < %s -march=nvptx64 -mcpu=sm_20 | FileCheck %s
```
[the remark this answers is not in the document]
Comment (AlexMaclean) [reply llvm/test/CodeGen/NVPTX/bswap.ll:2]: Sounds good, fixed

**Review by Artem-B — APPROVED** [review]

**Pull request merged**